datafusion/sql/src/statement.rs

```diff
@@ -1239,8 +1240,9 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         let ordered_exprs =
             self.build_order_by(order_exprs, &df_schema, &mut planner_context)?;
 
-        // External tables do not support schemas at the moment, so the name is just a table name
-        let name = TableReference::bare(name);
+        let idents: Vec<Ident> = name.split('.').map(Ident::from).collect();
+        let name = object_name_to_table_reference(ObjectName(idents), false)
+            .unwrap_or(TableReference::bare(name));
```

Comment:
```suggestion
        let name = self.object_name_to_table_reference(name)?;
```
We should respect the `enable_ident_normalization` option here